evalml/pipelines/pipeline_base_meta.py

```diff
@@ -15,14 +16,18 @@ def check_for_fit(cls, method):
             It raises an exception if `False` and calls and returns the wrapped method if `True`.
         """
         @wraps(method)
-        def _check_for_fit(self, X=None, y=None):
+        def _check_for_fit(self, X=None, y=None, objective=None):
             klass = type(self).__name__
             if not self._is_fitted:
                 raise PipelineNotYetFittedError(f'This {klass} is not fitted yet. You must fit {klass} before calling {method.__name__}.')
             elif X is None and y is None:
                 return method(self)
             elif y is None:
                 return method(self, X)
+            # For time series classification pipelines, predict will take X, y, objective
+            elif len(inspect.getfullargspec(method).args) == 4:
+                return method(self, X, y, objective)
+            # For other pipelines, predict will take X, y or X, objective
```

Comment:
Yea it's correct but confusing 😅 .

For non-timeseries classification pipelines, the arguments will be `X` and `objective` so the `y` in this `else` is actually the `objective` the user passed in!

I left this (comment)[https://github.com/alteryx/evalml/pull/1528#discussion_r540318731] earlier  but I guess it got overwritten when I pushed new changes:

> The problem is that TimeSeriesBinaryClassification.predict accepts X, y and objective. All other pipeline classes accept X and y (TimeSeriesRegressionPipeline) or X and objective (BinaryClassificationPipeline, MulticlassClassificationPipeline) or X (ClassificationPipeline).
> 
> I think adding this extra check for the number of arguments in predict is the lowest-touch way of handling the complexity (as opposed to adding another metaclass) especially since this class is not user-facing and we don't expect users to define their own pipeline base classes.
> 
> That being said, maybe we can refactor this metaclass definition to make it a little clearer what's going on?